src/code/z_lights.c

```diff
@@ -1,190 +1,398 @@
 }
 
-void Lights_InitType2PositionalLight(LightInfoPositional* info, s16 posX, s16 posY, s16 posZ, u8 red, u8 green, u8 blue,
-                                     s16 radius) {
-    Lights_InitPositionalLight(info, posX, posY, posZ, red, green, blue, radius, 2);
+void Lights_PointGlowSetInfo(LightInfo* info, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius) {
+    Lights_PointSetInfo(info, x, y, z, r, g, b, radius, LIGHT_POINT_GLOW);
 }
 
-void Lights_SetPositionalLightColorAndRadius(LightInfoPositional* info, u8 red, u8 green, u8 blue, s16 radius) {
-    info->params.red = red;
-    info->params.green = green;
-    info->params.blue = blue;
-    info->params.radius = radius;
+void Lights_PointSetColorAndRadius(LightInfo* info, u8 r, u8 g, u8 b, s16 radius) {
+    info->params.point.color[0] = r;
+    info->params.point.color[1] = g;
+    info->params.point.color[2] = b;
+    info->params.point.radius = radius;
 }
 
-void Lights_InitDirectional(LightInfoDirectional* info, s8 dirX, s8 dirY, s8 dirZ, u8 red, u8 green, u8 blue) {
-    info->type = 1;
-    info->params.dirX = dirX;
-    info->params.dirY = dirY;
-    info->params.dirZ = dirZ;
-    info->params.red = red;
-    info->params.green = green;
-    info->params.blue = blue;
+void Lights_DirectionalSetInfo(LightInfo* info, s8 x, s8 y, s8 z, u8 r, u8 g, u8 b) {
+    info->type = LIGHT_DIRECTIONAL;
+    info->params.dir.x = x;
+    info->params.dir.y = y;
+    info->params.dir.z = z;
+    info->params.dir.color[0] = r;
+    info->params.dir.color[1] = g;
+    info->params.dir.color[2] = b;
 }
 
-void Lights_MapperInit(LightMapper* mapper, u8 red, u8 green, u8 blue) {
-    mapper->ambient.l.col[0] = red;
-    mapper->ambient.l.colc[0] = red;
-    mapper->ambient.l.col[1] = green;
-    mapper->ambient.l.colc[1] = green;
-    mapper->ambient.l.col[2] = blue;
-    mapper->ambient.l.colc[2] = blue;
-    mapper->numLights = 0;
+// unused
+void Lights_Reset(Lights* lights, u8 r, u8 g, u8 b) {
+    lights->l.a.l.col[0] = lights->l.a.l.colc[0] = r;
+    lights->l.a.l.col[1] = lights->l.a.l.colc[1] = g;
+    lights->l.a.l.col[2] = lights->l.a.l.colc[2] = b;
+    lights->numLights = 0;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_lights/func_80079EFC.s")
+/*
+ * Draws every light in the provided Lights group.
+ *
+ * Note: Due to how Lights_Update is implemented, this will end up drawing every light in the buffer,
+ *       even if the light was not added by the system invoking this function.
+ */
+void Lights_Draw(Lights* lights, GraphicsContext* gfxCtxArg) {
+    Light* light;
+    s32 i;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
+
+    gfxCtx = gfxCtxArg;
+    Graph_OpenDisps(dispRefs, gfxCtx, "../z_lights.c", 339);
+
+    gSPNumLights(gfxCtx->polyOpa.p++, lights->numLights);
+    gSPNumLights(gfxCtx->polyXlu.p++, lights->numLights);
+
+    i = 0;
+    light = &lights->l.l[0];
+
+    while (i < lights->numLights) {
```

Comment:
Did you try a for loop here? It might work better if possible

Reply:
this loop was very picky and this is the only configuration that worked. roman and krim can both attest FeelsOkayMan

Reply:
gorcha, kk :)